go-controller/pkg/node/routemanager/route_manager.go

```diff
@@ -580,48 +319,30 @@ func subscribeNetlinkRouteEvents(stopCh <-chan struct{}) (bool, chan netlink.Rou
 	return true, routeEventCh
 }
 
-type NetLinkOps interface {
-	LinkList() ([]netlink.Link, error)
-	LinkByName(ifaceName string) (netlink.Link, error)
-	LinkByIndex(index int) (netlink.Link, error)
-	IsLinkNotFoundError(err error) bool
-	RouteDel(route *netlink.Route) error
-	RouteAdd(route *netlink.Route) error
-	RouteReplace(route *netlink.Route) error
-	RouteListFiltered(family int, filter *netlink.Route, filterMask uint64) ([]netlink.Route, error)
-}
-
-type defaultNetLinkOps struct {
+// RoutePartiallyEqual compares a limited set of route attributes.
+// The reason for not using the Equal method associated with type netlink.Route is because a user will only specify a limited
+// subset of fields but when we introspect routes seen on the system, other fields are populated by default and therefore
+// won't be equal anymore with user defined routes. Compare a limited set of fields that we care about.
+// Also, netlink.Routes Equal method doesn't compare MTU.
+func RoutePartiallyEqual(r, x netlink.Route) bool {
+	return r.LinkIndex == x.LinkIndex &&
+		ipNetEqual(r.Dst, x.Dst) &&
+		r.Src.Equal(x.Src) &&
+		r.Gw.Equal(x.Gw) &&
+		r.Table == x.Table &&
+		r.Flags == x.Flags &&
+		r.MTU == x.MTU
 }
 
-func (defaultNetLinkOps) LinkList() ([]netlink.Link, error) {
-	return netlink.LinkList()
-}
-
-func (defaultNetLinkOps) LinkByName(ifaceName string) (netlink.Link, error) {
-	return netlink.LinkByName(ifaceName)
-}
-
-func (defaultNetLinkOps) LinkByIndex(index int) (netlink.Link, error) {
-	return netlink.LinkByIndex(index)
-}
-
-func (defaultNetLinkOps) IsLinkNotFoundError(err error) bool {
-	return reflect.TypeOf(err) == reflect.TypeOf(netlink.LinkNotFoundError{})
-}
-
-func (defaultNetLinkOps) RouteDel(route *netlink.Route) error {
-	return netlink.RouteDel(route)
-}
-
-func (defaultNetLinkOps) RouteAdd(route *netlink.Route) error {
-	return netlink.RouteAdd(route)
-}
-
-func (defaultNetLinkOps) RouteReplace(route *netlink.Route) error {
-	return netlink.RouteReplace(route)
-}
-
-func (defaultNetLinkOps) RouteListFiltered(family int, filter *netlink.Route, filterMask uint64) ([]netlink.Route, error) {
-	return netlink.RouteListFiltered(family, filter, filterMask)
+// ipNetEqual returns true if both IPNet are equal
+func ipNetEqual(ipn1 *net.IPNet, ipn2 *net.IPNet) bool {
```

Comment:
should probably go in the common util package like https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/net_linux.go